crates/telio-traversal/src/endpoint_providers/stun.rs

```diff
@@ -384,6 +403,21 @@ impl<Wg: WireGuard, E: Backoff> State<Wg, E> {
             .await
     }
 
+    async fn transition_to_wait_for_wg(&mut self) {
+        self.stun_state = StunState::WaitingForWg;
+        self.current_timeout = PinnedSleep::new(STUN_TIMEOUT, ());
+
+        if self.is_wg_ready().await {
+            self.transition_to_searching_for_server();
```

Comment:
Why is a function which is called "transition to wait for WG" doing a transition to searching for server?

Reply:
You're right, this is probably not a good idea